Allow `noServerMethods` option and otherwise throw error upon loading…

brettz9 opened the pull request:
… store names which conflict with db.js methods; add test and docs

To address issue #136

aaronpowell commented on tests/specs/open-db.js:
```diff
@@ -56,6 +56,20 @@
+            }).then(function (s) {
+                spec.server = s;
+                expect(spec.server).toBeDefined();
+                expect(spec.server.query).toEqual(jasmine.any(Function));
```
I don't think the `expect` really provides much benefit. It'd be better to setup a schema and assert that the `server.<tableName>` doesn't exist.

brettz9 replied:
Gotcha, makes sense.

aaronpowell commented on tests/specs/open-db.js:
```diff
@@ -56,6 +56,20 @@
         });
 
+        it('should allow avoidance of server methods', function (done) {
+            var spec = this;
```
There's no need to capture the `this` state in the test.

aaronpowell commented on tests/specs/open-db.js:
```diff
@@ -56,6 +56,20 @@
+                version: 1,
+                noServerMethods: true
+            }).then(function (s) {
+                spec.server = s;
```
You don't need to expose the server up so there's no need to do that

brettz9 commented:
I've added a commit to handle the case that you suggested (and I was mistakenly checking for conflicts against the db name instead of against the object store). I did still think it was useful (via the second commit) to have a check to confirm that `query` was a function (as opposed to an object) as it ensures that `query` is still available as a server method (and not an object store), but I also added a test specifically to check that a valid store name was not present with `noServerMethods` added.

The first commit also changes things to self-close the connection since otherwise the db might remain blocked.

Let me know if you want me to revert the other changes in the first commit to reject instead of throw--I thought it might provide a clearer flow.

aaronpowell commented on tests/specs/open-db.js:
```diff
@@ -56,6 +56,49 @@
+        });
+
+        it('should not add stores to server using noServerMethods', function (done) {
+            var spec = this;
```
No need to capture this here

brettz9 replied:
Why not confirm that an error is thrown?

aaronpowell commented on tests/specs/open-db.js:
```diff
@@ -56,6 +56,49 @@
+                    }
+                }
+            }).then(function (s) {
+                spec.server = s;
```
don't need this here, just assert off `s`

brettz9 replied:
I set spec.server so that the `afterEach` had the server to close...Otherwise, the next test got blocked...

aaronpowell replied:
Ah good point, forgot that that needs to be done!

brettz9 commented:
Are you ok with my further modifications here?
